### ISSUE TITLE
New test failures on Level0

### ISSUE THREAD

**Issue by linehill** [CHIP-SPV/chip-spv#99]
The following test cases fail on Level0 after faec3a6969a7e3488d7c24ac96bfa74865a18983:
```
	 64 - cuda-simpleCallback (Subprocess aborted)
	 72 - cuda-qrng (Failed)
	 77 - cuda-FDTD3d (SEGFAULT)
```

**Comment by pvelesko** [comment]
After #101: 
```
	346 - activemask (Failed)
	415 - cuda-simpleCallback (Timeout)
	423 - cuda-qrng (Failed)
```

415 is known, 346 is expected I believe?

**Comment by franz** [comment]
@pvelesko not quite, activemask test is supposed to fail with "unsupported" message, but currently it fails because `cuspv` was changed to call `hipcc`, and hipcc can't find the clang binary (`hipcc: clang++ not found. Please add clang++ to PATH or set HIP_CLANG_PATH`)

415 will be disabled in #95, i don't know about 423

**Comment by pvelesko** [comment]
> and hipcc can't find the clang binary (hipcc: clang++ not found. Please add clang++ to PATH or set HIP_CLANG_PATH)

activemask is passing for me with the last 2 PRs now merged. Can you confirm? Did you implement the symlink for `clang++` ?

**Comment by pvelesko** [comment]
Here is a total list of tests fail for me when run `ctest --timeout 120 -j 8`

````
86% tests passed, 97 tests failed out of 707

Label Time Summary:
cuda        =  57.32 sec*proc (27 tests)
internal    = 353.86 sec*proc (79 tests)

Total Test time (real) = 184.03 sec

The following tests FAILED:
	 65 - Unit_hipMemcpy2D_SizeCheck (Failed)
	 66 - Unit_hipMemcpy2D_Negative (Failed)
	129 - Unit_hipMemset_SetMemoryWithOffset (Failed)
	130 - Unit_hipMemsetAsync_SetMemoryWithOffset (Failed)
	152 - Unit_hipMemset3DAsync_ConcurrencyMthread (Subprocess aborted)
	160 - Unit_hipMemPrefetchAsync_NonPageSz (Failed)
	197 - Unit_hipHostMalloc_NonCoherent (Failed)
	198 - Unit_hipHostMalloc_Coherent (Failed)
	221 - Unit_hipMemcpyAsync_hipMultiMemcpyMultiThreadMultiStream - float (Subprocess aborted)
	222 - Unit_hipMemcpyAsync_hipMultiMemcpyMultiThreadMultiStream - double (Subprocess aborted)
	227 - Unit_AnyAll_CompileTest (Failed)
	228 - Unit_ballot (Failed)
	231 - Unit_funnelshift (Failed)
	235 - Unit_syncthreads_and (Failed)
	236 - Unit_syncthreads_count (Failed)
	237 - Unit_syncthreads_or (Failed)
	254 - Unit_deviceFunctions_CompileTest___fmaf_ieee_rd_float (Failed)
	255 - Unit_deviceFunctions_CompileTest___fmaf_ieee_rn_float (Failed)
	256 - Unit_deviceFunctions_CompileTest___fmaf_ieee_ru_float (Failed)
	257 - Unit_deviceFunctions_CompileTest___fmaf_ieee_rz_float (Failed)
	318 - Unit_deviceFunctions_CompileTest___funnelshift_l_int (Failed)
	319 - Unit_deviceFunctions_CompileTest___funnelshift_lc_int (Failed)
	320 - Unit_deviceFunctions_CompileTest___funnelshift_r_int (Failed)
	321 - Unit_deviceFunctions_CompileTest___funnelshift_rc_int (Failed)
	374 - Unit_deviceFunctions_CompileTest_j0f_float (Failed)
	375 - Unit_deviceFunctions_CompileTest_j1f_float (Failed)
	376 - Unit_deviceFunctions_CompileTest_jnf_float (Failed)
	378 - Unit_deviceFunctions_CompileTest_lgammaf_float (Failed)
	379 - Unit_deviceFunctions_CompileTest_llrintf_float (Failed)
	380 - Unit_deviceFunctions_CompileTest_llroundf_float (Failed)
	386 - Unit_deviceFunctions_CompileTest_lrintf_float (Failed)
	387 - Unit_deviceFunctions_CompileTest_lroundf_float (Failed)
	396 - Unit_deviceFunctions_CompileTest_normcdff_float (Failed)
	397 - Unit_deviceFunctions_CompileTest_normcdfinvf_float (Failed)
	404 - Unit_deviceFunctions_CompileTest_rintf_float (Failed)
	405 - Unit_deviceFunctions_CompileTest_rnorm3df_float (Failed)
	406 - Unit_deviceFunctions_CompileTest_rnorm4df_float (Failed)
	464 - Unit_deviceFunctions_CompileTest_j0_double (Failed)
	465 - Unit_deviceFunctions_CompileTest_j1_double (Failed)
	466 - Unit_deviceFunctions_CompileTest_jn_double (Failed)
	468 - Unit_deviceFunctions_CompileTest_lgamma_double (Failed)
	469 - Unit_deviceFunctions_CompileTest_llrint_double (Failed)
	470 - Unit_deviceFunctions_CompileTest_llround_double (Failed)
	476 - Unit_deviceFunctions_CompileTest_lrint_double (Failed)
	477 - Unit_deviceFunctions_CompileTest_lround_double (Failed)
	487 - Unit_deviceFunctions_CompileTest_normcdf_double (Failed)
	488 - Unit_deviceFunctions_CompileTest_normcdfinv_double (Failed)
	494 - Unit_deviceFunctions_CompileTest_rint_double (Failed)
	496 - Unit_deviceFunctions_CompileTest_rnorm3d_double (Failed)
	497 - Unit_deviceFunctions_CompileTest_rnorm4d_double (Failed)
	515 - Unit_deviceFunctions_CompileTest_yn_double (Failed)
	519 - Unit_deviceFunctions_CompileTest_max_int (Failed)
	520 - Unit_deviceFunctions_CompileTest_min_int (Failed)
	524 - Unit_hipStreamCreateWithPriority (Failed)
	527 - Unit_hipStreamGetPriority_Negative (Failed)
	528 - Unit_hipStreamGetPriority_default (Failed)
	529 - Unit_hipStreamGetPriority_high (Failed)
	530 - Unit_hipStreamGetPriority_higher (Failed)
	531 - Unit_hipStreamGetPriority_low (Failed)
	532 - Unit_hipStreamGetPriority_lower (Failed)
	534 - Unit_hipMultiStream_multimeDevice (Timeout)
	535 - Unit_hipStreamAddCallback_MultipleThreads (SEGFAULT)
	538 - Unit_hipStreamAddCallback_WithCreatedStream (SEGFAULT)
	540 - Unit_hipStreamCreateWithPriority_FunctionalForAllPriorities (Failed)
	541 - Unit_hipStreamCreateWithPriority_MulthreadDefaultflag (Failed)
	542 - Unit_hipStreamCreateWithPriority_MulthreadNonblockingflag (Failed)
	543 - Unit_hipStreamCreateWithPriority_NegTst (Failed)
	544 - Unit_hipStreamCreateWithPriority_ValidateWithEvents (Failed)
	550 - Unit_hipEvent (Failed)
	556 - Unit_hipEventIpc (Failed)
	557 - Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_Negative (Failed)
	558 - Unit_hipOccupancyMaxActiveBlocksPerMultiprocessor_rangeValidation (Failed)
	560 - Unit_hipOccupancyMaxPotentialBlockSize_rangeValidation (Failed)
	575 - Unit_hipDeviceGetCacheConfig_FuncTst (Failed)
	586 - Unit_hipGetDeviceFlags_FuncTst (Failed)
	594 - Unit_hipStreamPerThread_Basic (Failed)
	595 - Unit_hipStreamPerThread_StreamQuery (Subprocess aborted)
	596 - Unit_hipStreamPerThread_StreamSynchronize (Subprocess aborted)
	597 - Unit_hipStreamPerThread_StreamGetPriority (Failed)
	598 - Unit_hipStreamPerThread_StreamGetFlags (Failed)
	600 - Unit_hipStreamPerThread_MemcpyAsync (Failed)
	601 - Unit_hipStreamPerThread_EventRecord (Failed)
	602 - Unit_hipStreamPerThread_EventSynchronize (Failed)
	606 - Unit_hipGraphAddEmptyNode_Functional (Failed)
	607 - Unit_hipGraphAddDependencies_Functional (Failed)
	608 - Unit_hipGraph_BasicFunctional (Failed)
	609 - Unit_hipGraph_SimpleGraphWithKernel (Failed)
	610 - Unit_hipGraphAddMemcpyNode_Functional (Failed)
	611 - ABM_AddKernel_MultiTypeMultiSize - int (Failed)
	612 - ABM_AddKernel_MultiTypeMultiSize - long (Failed)
	613 - ABM_AddKernel_MultiTypeMultiSize - float (Failed)
	614 - ABM_AddKernel_MultiTypeMultiSize - long long (Failed)
	615 - ABM_AddKernel_MultiTypeMultiSize - double (Failed)
	618 - Stress_hipMemcpy_multiDevice-AllAPIs - int (Failed)
	619 - Stress_hipMemcpy_multiDevice-AllAPIs - size_t (Failed)
	620 - Stress_hipMemcpy_multiDevice-AllAPIs - long double (Failed)
	691 - cuda-simpleCallback (SEGFAULT)
````

**Comment by franz** [comment]
testing with updated main @ bb4304266d32c5 it's still failing (and not with the expected "unsupported" message):

```
622: clang-15: warning: -lCHIP: 'linker' input unused [-Wunused-command-line-argument]
622: clang-15: warning: -lOpenCL: 'linker' input unused [-Wunused-command-line-argument]
622: clang-15: warning: -lze_loader: 'linker' input unused [-Wunused-command-line-argument]
622: clang-15: warning: -Wl,-rpath,/home/michal/0/install/lib: 'linker' input unused [-Wunused-command-line-argument]
622: clang-15: error: cannot find HIP runtime; provide its path via '--rocm-path', or pass '-nogpuinc' to build without HIP runtime
622: clang-15: error: cannot find HIP device library for 'spirv64' target; provide its path via '--hip-path' or '--hip-device-lib-path', or pass '-nogpulib' to build without HIP device library
1/1 Test #622: activemask .......................***Failed  Required regular expression not found.Regex=[error: '__activemask' is unavailable: unsupported in CHIP-SPV
```

... cuspv / hipcc seems still broken here. Do i need to update HIP ? i'm currently on ad8d67fd

**Comment by pvelesko** [comment]
Hmm not sure what is going on here. Seems like a problem with the test? Have you rebuilt the tests? I'm on the same HIP checkout:

````
pvelesko@iris18:~/space/CHIP-SPV/build> ctest -R activemask -V
UpdateCTestConfiguration  from :/home/pvelesko/space/CHIP-SPV/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/pvelesko/space/CHIP-SPV/build/DartConfiguration.tcl
Test project /home/pvelesko/space/CHIP-SPV/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 622
    Start 622: activemask

622: Test command: /usr/bin/bash "-c" "/gpfs/jlse-fs0/users/pvelesko/CHIP-SPV/build/bin/cuspvc -fsyntax-only activemask.cu 2>&1"
622: Test timeout computed to be: 10000000
622: clang-14: warning: -lCHIP: 'linker' input unused [-Wunused-command-line-argument]
622: clang-14: warning: -lOpenCL: 'linker' input unused [-Wunused-command-line-argument]
622: clang-14: warning: -lze_loader: 'linker' input unused [-Wunused-command-line-argument]
622: clang-14: warning: -Wl,-rpath,/gpfs/jlse-fs0/users/pvelesko/CHIP-SPV/install/lib: 'linker' input unused [-Wunused-command-line-argument]
622: clang-14: warning: argument unused during compilation: '-L/gpfs/jlse-fs0/users/pvelesko/CHIP-SPV/install/lib' [-Wunused-command-line-argument]
622: activemask.cu:5:11: error: '__activemask' is unavailable: unsupported in CHIP-SPV.
622:   *Mask = __activemask();
622:           ^
622: /gpfs/jlse-fs0/users/pvelesko/CHIP-SPV/include/hip/spirv_hip_devicelib.hh:570:17: note: '__activemask' has been explicitly marked unavailable here
622: EXPORT unsigned __activemask()
622:                 ^
622: 1 error generated when compiling for host.
1/1 Test #622: activemask .......................   Passed    0.54 sec

The following tests passed:
	activemask

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.62 sec
pvelesko@iris18:~/space/CHIP-SPV/build>

````

**Comment by franz** [comment]
what's going on is that

1) `cuspv` calls `hipcc`
2) `hipcc` loads `hipvars.pm`
3) `hipvars.pm` loads `lib/.hipInfo`

.. and `.hipInfo` contains installation paths, not build paths. So if you have chip-spv installed then it picks that up. But `cuspv` is supposed to be a build-time script, so this is incorrect (for install-time script we have `cuspv-install`). We need either another `.hipInfo` with build paths, or use the old way of calling clang directly from cuspv.

**Comment by linehill** [comment]
FYI, I'm making a fix for .hipInfo issue by creating separate version of the file for build and install directory.

**Comment by linehill** [comment]
> ... (for install-time script we have `cuspv-install`) ...

Please note that the script is renamed to cuspv**c** during installation.

**Comment by linehill** [comment]
Made a fix candidate (#108) for cuspvc working in the build directory.

**Comment by pvelesko** [comment]
fixed in #123
